test-suite/Math/NumberTheory/Primes/QuadraticSieveTests.hs

```diff
@@ -7,23 +7,36 @@ import Test.Tasty.QuickCheck as QC
 import Math.NumberTheory.TestUtils ()
 import Math.NumberTheory.Primes
 import Math.NumberTheory.Primes.Factorisation.QuadraticSieve
+import qualified Debug.Trace
+
+trace :: String -> a -> a
+trace = if debug then Debug.Trace.trace else const id
+
+debug :: Bool
+debug = False
+
+checkSquares :: Large Int -> Large Int -> Bool
+checkSquares (Large i) (Large j)
+  | p == 2 || q == 2 || p == q = True
+  | otherwise                  = (x * x - y * y) `mod` n == 0
+  where
+    (x, y) = trace ("Number: " ++ show n) $ head $ findSquares n $ autoConfig n
+    n = p * q
+    p = toInteger . unPrime . nextPrime $ i `mod` 100000000
+    q = toInteger . unPrime . nextPrime $ j `mod` 100000000
```

Comment:
Is it still desirable to reduce modulo 100000000 here?

Reply:
Yes, but the tests will take longer